giftcertificate/service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -1,56 +1,96 @@
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
-import java.util.List;
-import java.util.Objects;
+import java.util.*;
 
 @Service
 public class UserServiceImpl implements UserService {
 
     private static final String KEY_USER_ID_NOT_EXIST = "user_id_not_exist";
-    /**
-     * TagJDBCTemplate is used for operations with User
-     */
+    private static final String USER_ROLE = "ROLE_USER";
+    private static final String ADMIN_AUTHORITY = "ROLE_ADMIN";
+    private static final String FORBIDDEN="get_user_forbidden";
+
     @Autowired
     private UserDAO userDAO;
 
-    /**
-     * Read user by id
-     *
-     * @param id id of user
-     * @return User
-     * @throws IdNotExistServiceException if user with passed id is not exist
-     */
+    @Autowired
+    private RoleDAO roleDAO;
+
+    @Autowired
+    private PasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
+    private ModelMapper modelMapper;
+
     @Override
+    @Secured("ROLE_USER")
     public User getUser(long id) throws IdNotExistServiceException {
-        User user;
-        user = userDAO.getUser(id);
-        if (Objects.isNull(user)) {
+        Optional<User> user;
+        JwtUser jwtUser = (JwtUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+       if(RoleUtil.hasAdminAuthority(jwtUser)){
+           user = userDAO.findById(id);
+       }else{
+           if(id==jwtUser.getId()){
+               user=userDAO.findById(jwtUser.getId());
+           }else {
+               throw new JwtAuthenticationException(FORBIDDEN);
+           }
+       }
+        if (!user.isPresent()) {
             throw new IdNotExistServiceException(KEY_USER_ID_NOT_EXIST);
         }
-        return user;
+        return user.get();
     }
 
-    /**
-     * Read users from DB
-     *
-     * @param page number of page
-     * @param size size of page
-     * @return List of users
-     * @throws PaginationException if page equals zero
-     */
+
     @Override
+    @Secured("ROLE_ADMIN")
     public List<User> getUsers(Integer page, Integer size) throws PaginationException {
         page = PaginationUtil.checkPage(page);
         size = PaginationUtil.checkSizePage(size);
-        return userDAO.getUsers(page, size);
+        Pageable pageable = PageRequest.of(page - 1, size);
+        return userDAO.findAll(pageable).getContent();
+    }
+
+    @Override
+    @Transactional
+    public User register(UserDto userDto) {
+        User user = modelMapper.map(userDto, User.class);
+        Role role = roleDAO.findRoleByName(USER_ROLE);
+        List<Role> roles = new ArrayList<>();
+        roles.add(role);
+        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
+        user.setRoles(roles);
+        User reggisteredUser = userDAO.save(user);
```

Comment:
you don't need to assign a variable for that, you never use this User anywhere further